includes/class-send-list.php

```diff
@@ -0,0 +1,213 @@
+		return [
+			'type'                 => 'object',
+			'additionalProperties' => false,
+			'properties'           => [
+				// The slug of the ESP for which this list or sublist originates.
+				'provider'    => [
+					'name'     => 'provider',
+					'type'     => 'string',
+					'required' => true,
+					'enum'     => Newspack_Newsletters::get_supported_providers(),
+				],
+				// The type of list. Can be either 'list' or 'sublist'. If the latter, must specify a `parent` property.
+				'type'        => [
+					'name'     => 'type',
+					'type'     => 'string',
+					'required' => true,
+					'enum'     => [
+						'list',
+						'sublist',
+					],
+				],
+				// The type of entity this list or sublist is associated with in the ESP. Controls which ESP API endpoints to use to fetch/update.
+				'entity_type' => [
+					'name'     => 'entity_type',
+					'type'     => 'string',
+					'required' => true,
+				],
+				// The ID of the list or sublist as identified in the ESP.
+				'id'          => [
+					'name'     => 'id',
+					'type'     => 'string',
+					'required' => true,
+				],
+				'value'       => [
+					'name'     => 'value',
+					'type'     => 'string',
+					'required' => false,
+				],
+				// The name of the list or sublist as identified in the ESP.
+				'name'        => [
+					'name'     => 'name',
+					'type'     => 'string',
+					'required' => true,
+				],
+				'label'       => [
+					'name'     => 'label',
+					'type'     => 'string',
+					'required' => false,
+				],
+				// If the list is also a Subscription List, it could have a locally edited name.
+				'local_name'  => [
+					'name'     => 'local_name',
+					'type'     => 'string',
+					'required' => false,
+				],
+				// If available, the number of contacts associated with this list or sublist.
+				'count'       => [
+					'name'     => 'count',
+					'type'     => 'integer',
+					'required' => false,
+				],
+				// If this Send_List is a sublist, this property must indicate the ID of the parent list.
+				'parent_id'   => [
+					'name'     => 'parent_id',
+					'type'     => 'string',
+					'required' => false,
+				],
+				// If it can be calculated, the URL to view this list or sublist in the ESP's dashboard.
+				'edit_link'   => [
+					'name'     => 'edit_link',
+					'type'     => 'string',
+					'required' => false,
+				],
+			],
+		];
+	}
+
+	/**
+	 * Get the Send_List configuration.
+	 *
+	 * @return array
+	 */
+	public function get_config() {
+		$schema = self::get_config_schema();
+		$config  = [];
+		foreach ( $schema['properties'] as $key => $property ) {
+			$config[ $key ] = $this->get( $key ) ?? null;
+		}
+
+		return array_filter( $config );
+	}
+
+	/**
+	 * Get a specific property's value.
+	 *
+	 * @param string $key The property to get.
+	 *
+	 * @return mixed The property value or null if not set/not a supported property.
+	 */
+	public function get( $key ) {
+		return $this->{ $key } ?? null;
+	}
```

Comment:
Following the steps from the `Subscription_List`, I wonder if it's worth being more explicit using named methods like `get_provider()` rather than `get( 'provider' )`.

I understand it requires more boilerplate but the encapsulation might be better in the long run.

Maybe @leogermani could share his thoughts?

Reply:
I agree. 7f4906e adds the following methods:

- `get_errors`
- `get_provider`
- `get_id`
- `get_type`
- `get_entity_type`
- `get_name`
- `get_local_name` (this isn't used anywhere currently, but I included local name so we can have the option to inherit the local name from the `Subscription_List` created from the same entity)
- `get_label`
- `get_value`
- `set_name`
- `set_local_name`

This makes it clearer which properties you might need to get and which can be set after the object has been constructed. e.g. you can't reset `id` since it's a required property.